### DISCUSSION TITLE
FYI: https://github.com/jinnovation/kele.el

### DISCUSSION THREAD

**Discussion by hab25** [eshelyaron/kubed#3] [General]
In case you already knew about it, could you please share the reason(s) for why you preferred creating your own package instead of using or contributing to `kele`? 

Such a comparison would be useful even if you don't intend to keep updating it as both the packages update; e.g. see https://github.com/joaotavora/eglot/blob/db91d58374627a195b731a61bead9b4f84a7e4bc/README.md?plain=1#L208 .

**Comment by eshelyaron** [comment]
Hi @hab25, thanks for reaching out!

The main reason for creating Kubed despite the existence of other
Kubernetes packages, is that Kubed is intended to be added to core
Emacs, or at least allow for that option.  Kubed has no external
dependencies, and its copyright belongs to the FSF.

I think that's important because I would like Emacs to provide great
user experience for professional developers, and working with Kubernetes
is very common nowadays.

There are also quite a few differences in terms of the features each
package provides.  I would indeed like to include a comparison with some
of the alternatives, to help people figure out which package is likely
to suit them best.  However, that's not my top priority at the moment.
If someone writes such a comparison, or even just a few notes, I'd love
to incorporate it or refer to it in the docs.

**Reply by jinnovation** [reply]
@eshelyaron: I've added a starting comparison between Kele and Kubed [here](https://jonathanj.in/kele.el/explanations/comparisons/#kubed). **Please** let me know if I've misrepresented anything about Kubed and I will happily revise!

**Reply by eshelyaron** [reply]
Hi @jinnovation, thanks for that!
That mostly looks fine, I think one important difference is that Kubed avoids 3rd party dependencies.  This makes it possible to add Kubed to core Emacs, if we'll decide to do that someday.
Also, I wouldn't say that Kubed "[works] through a centralized Transient-based interface", indeed there are transient menus, but I think the most common way to use Kubed is saying (e.g.) `kubed-list-pods` and working with the tabulated list interface.
In terms of functionality, there are also a couple things that I think are exclusive to Kubed (not sure though), namely bookmarking and good Tramp integration.  In general Kubed tries to "make the most" out of core Emacs integration.

**Reply by jinnovation** [reply]
>  I think one important difference is that Kubed avoids 3rd party dependencies. This makes it possible to add Kubed to core Emacs, if we'll decide to do that someday.

Agree. Will update to call this out + the focus on core Emacs integrations.

> namely bookmarking and good Tramp integration

Likewise.

**Reply by jinnovation** [reply]
@eshelyaron: Updated [here](https://github.com/jinnovation/kele.el/commit/114fc1c3a2a5ba68732510cdf696a0df504cc34f). Would love your thoughts.

**Reply by eshelyaron** [reply]
@jinnovation LGTM, thanks!
